src/components/sbb-calendar/sbb-calendar.tsx

```diff
@@ -0,0 +1,420 @@
+        {this._dateAdapter.getYear(d)}
+      </span>
+    );
+  }
+
+  /** Checks if date is within the min-max range. */
+  private _isDayInRange(date: string): boolean {
+    if (!this._min && !this._max) {
+      return true;
+    }
+    const isBeforeMin =
+      this._dateAdapter.isValid(this._min) &&
+      this._dateAdapter.compareDate(this._min, new Date(date)) > 0;
+    const isAfterMax =
+      this._dateAdapter.isValid(this._max) &&
+      this._dateAdapter.compareDate(this._max, new Date(date)) < 0;
+    return !(isBeforeMin || isAfterMax);
+  }
+
+  /** Emits the selected date and sets it internally. */
+  private _selectDate(day: string, event: Event): void {
+    event.stopImmediatePropagation();
+    if (this._selected !== day) {
+      this._selected = day;
+      this.dateSelected.emit(new Date(day));
+    }
+  }
+
+  private _handleKeyboardEvent(event, day: Day): void {
+    if (isArrowKeyOrPageKeysPressed(event)) {
+      event.preventDefault();
+    }
+    const index = this._days.findIndex((e: HTMLButtonElement) => e === event.target);
+    handleKeyboardEvent(event, index, this._days, day)?.focus();
+  }
+
+  /** Goes to the previous month. */
+  private _goToMonth(event: Event, months: number): void {
+    event.stopImmediatePropagation();
+    this._assignActiveDate(this._dateAdapter.addCalendarMonths(this._activeDate, months));
+    this._init();
+  }
+
+  /** Checks if the "previous month" button is enabled. */
+  private _previousMonthEnabled(): boolean {
+    if (!this._min) {
+      return false;
+    }
+    const prevMonth = this._dateAdapter.clone(this._activeDate);
+    prevMonth.setDate(0);
+    return this._dateAdapter.compareDate(prevMonth, this._min) < 0;
+  }
+
+  /** Checks if the "next month" button is enabled. */
+  private _nextMonthEnabled(): boolean {
+    if (!this._max) {
+      return false;
+    }
+    const nextMonth = this._dateAdapter.addCalendarMonths(this._activeDate, this._wide ? 2 : 1);
+    nextMonth.setDate(1);
+    return this._dateAdapter.compareDate(nextMonth, this._max) > 0;
+  }
+
+  private _assignActiveDate(date: Date): void {
+    if (this._min && this._dateAdapter.compareDate(this._min, date) > 0) {
+      this._activeDate = this._min;
+      return;
+    }
+    if (this._max && this._dateAdapter.compareDate(this._max, date) < 0) {
+      this._activeDate = this._max;
+      return;
+    }
+    this._activeDate = date;
+  }
+
+  private _getFirstFocusable(): HTMLButtonElement {
+    let firstFocusable =
+      this._element.shadowRoot.querySelector('.sbb-datepicker__day-selected') ??
+      this._element.shadowRoot.querySelector('.sbb-datepicker__day-today');
+    if (!firstFocusable || (firstFocusable as HTMLButtonElement)?.disabled) {
+      firstFocusable = Array.from(
+        this._element.shadowRoot.querySelectorAll('.sbb-datepicker__day')
+      ).find((e) => !(e as HTMLButtonElement).disabled);
```

Comment:
I guess it can be shortened?
```suggestion
        this._element.shadowRoot.querySelector('.sbb-datepicker__day:not([disabled]')
      ));
```